src/main/java/com/nha/abdm/wrapper/hrp/serviceImpl/LogsTableService.java

```diff
@@ -124,11 +165,37 @@ public List<CareContextBuilder> getSelectedCareContexts(String linkRefNumber, Li
 
         return null;
     }
-    public String getStatus(JsonNode data) {
-        RequestLogs existingRecord=logsRepo.findByClientRequestId(data.get("requestId").asText());
+    public String getStatus(ResponseHelper data) {
+        RequestLogs existingRecord=logsRepo.findByClientRequestId(data.getRequestId());
         if(existingRecord!=null){
             return existingRecord.getResponse().toString();
         }
         return "Record failed but stored in database";
     }
+        public void setStatus(OnAddCareContextResponse data) {
```

Comment:
The gateway response should be captured in raw and not the status alone. And here only the response should be captured. The responsibility of updating the `Patient` table with the `careContext` linking should be given to the service layer.

Reply:
Dumping the on-add-contexts into db